patchview/gitdiff

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/python3
+#-*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2019 Sérgio Basto <sergio@serjux.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+import os
+import sys
+import argparse
+from subprocess import Popen, PIPE
```

Comment:
This script feels like it might be a better fit for shell rather than Python. Is there a strong reason for using Python here?

If it's just option parsing, maybe use `--` as a separator between git and patchview args, and a loop with `shift` to build up args for each?

Reply:
yes, it's just option parsing, python argparse [1] allow we have positional arguments ( git_args and  patchview_args)  We can give args to git and args to patchview , one common example : `gitdiff devel..HEAD -F5` is the same of `git diff devel..HEAD | patchview -F5`
The Python libraries os, sys, argparse and subprocess are included on Python 3 core , but yes we need have Python in the system to run the scripts .

Anyway these commit, can be commit in a second phase , if the other commits are accepted it is already a very good progress.


[1]
https://docs.python.org/3/library/argparse.html

Reply:
OK.